client/my-sites/hosting/staging-site-card/use-staging-sync.ts

```diff
@@ -0,0 +1,64 @@
+import { useMutation, UseMutationOptions, useIsMutating } from '@tanstack/react-query';
+import { useCallback, useMemo } from 'react';
+import wp from 'calypso/lib/wp';
+
+type MutationVariables = number;
+
+interface MutationResponse {
+	message: string;
+}
+
+interface MutationError {
+	code: string;
+	message: string;
+}
+
+export const PUSH_TO_STAGING = 'push-to-staging-site-mutation-key';
+
+const pushSyncMutation =
+	( sitePath: string, key: string ) =>
+	< X, Y, Z, W >( firstSiteId: number, options: UseMutationOptions< X, Y, Z, W > ) => {
+		const mutation = useMutation( {
+			mutationFn: async ( otherSiteId ) =>
+				wp.req.post( {
+					path: `/sites/${ firstSiteId }/${ sitePath }/${ otherSiteId }`,
```

Comment:
Would `source_site_id` and `destination_site_id` a clearer names?

Reply:
Because in the backend there is the notion of source and target, I'll name them as such. Thanks for your suggestion!